proposed/2021/MachineReadableOutput.md

```diff
@@ -0,0 +1,797 @@
+# Machine readable JSON output for dotnet list package
+
+* Status: **In Review**
+* Author: [Erick Yondon](https://github.com/erdembayar)
+* GitHub Issue [7752](https://github.com/NuGet/Home/issues/7752)
+
+## Problem background
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+Many organization are required by [regulation](https://www.whitehouse.gov/briefing-room/presidential-actions/2021/05/12/executive-order-on-improving-the-nations-cybersecurity/) to audit packages that they're using in repository.
+Currently there's no easy way produce Software Bill of Material (SBOM) output which can be consumed by another auditing system or historic keeping.
+
+* Parse-friendly output. Other PMP like Npm already have it(`npm ls --parseable` and `npm ls --json`).
+* Useful for CI/CD auditing(compliance, security ..)
+  * Produce Software Bill of Material (SBOM) (compliance, historic keeping)
+  * Enhancing Software Supply Chain Security
+    * check any vulnerable packages
+    * check any deprecated packages
+    * Check any outdated packages
+  * Check license compliance
+  * Resolve dependency issue (detect duplicate packages, detect new dependency introduced or existing one removed etc ..)
+  * Making the output machine-readable unlocks additional tooling and automation scenarios in CI/CD pipeline above scenarios.
+
+## Who are the customers
+
+Anyone (government/private enterprises, security experts, individual contributors) who wants to consume `dotnet list package` output for auditing tool or historic keeping, CI/CD orchestrating.
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+
+### --json option
+
+Ability to use new `--json` option for all `dotnet list package` commands to ensure JSON-formatted output is emitted to the console.
```

Comment:
> does that also imply that other formats than json and text can be used?

Only if `dotnet list package -h` says there are options other than text and json, which is not being proposed in this spec. But it's clear and obvious how extra formats could be added in the future, and would allow the command line parsing library to handle the error handling when customers provide more than one, rather than having to implement that error handling ourselves.

Reply:
I just wanted to +1 that. 

We should do `--format <arg>`

Reply:
Chiming in with my +1 on the `--format <blah>` or `--output <blah>` model, as that model is open to extension, already modeled quite well in prominent examples like the `az` CLI for Azure and the `aws` CLI for AWS. Slight preference for `--output`, because that terminology has been established by those tools. This proposal is the first such output proposal for the core .NET CLI tools, and so whatever we land on here is going to be what all other tools have to use, as we spread JSON/etc output to the rest of the CLI.

Reply:
@baronfel 
I favor `--format <blah>`. Because I can see `--output` is already used for another popular dotnet command.

![image](https://user-images.githubusercontent.com/8766776/145644723-a6e9227c-378a-4238-8c86-b0bec020e26a.png)

Reply:
@erdembayar good catch, I edited my comment before refreshing the page and seeing your response